instrumentation/ktor-1.0/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v1_0/IpAddressUtil.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.ktor.v1_0
+
+import java.util.regex.Pattern
+
+// Source: Regular Expressions Cookbook 2nd edition - 8.17.
+// Matching IPv6 Addresses
+private val ipv6 = Pattern.compile( // Non Compressed
```

Comment:
Just a pointer to Netty's isIpAddress methods

https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/NetUtil.java#L270

There're probably faster than regex, but also tons of code so not worth it but for reference in case it seems useful

Reply:
good to know, could be worth vendoring in to instrumentation-api at some point if we find it useful on more instrumentations

Reply:
Also: we already have a ipv6 parser in [`JdbcConnectionUrlParser`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/internal/JdbcConnectionUrlParser.java#L119), so there are now 2 instrumentations that would benefit from it.